error in installing dlip: ‘WebPFree’ was not declared in this scope

<!-- Delete the above instructions and then provide a general summary of the issue in the Title above.  Then fill out the template below.  FAILURE TO FILL OUT EVERY PART WILL RESULT IN YOUR ISSUE BEING CLOSED -->

## Expected Behavior
<!--- Tell us what should happen.  What were you doing?  What part of dlib are you using?  What do you think should happen? -->

When I install the dlib by following the documentation, I faced the issue as below



## Current Behavior
<!--- Tell us what happens instead of the expected behavior.  If you get an error, include the entire error message in the bug report.  DO NOT POST SCREEN SHOTS.  Paste in the text instead.   If the issue is some kind of build problem, include the entire CMake output along with the error message. -->

aibrain@aibrain-VirtualBox:~/lib/dlib/build$ cmake --build .
[  1%] Building CXX object dlib/CMakeFiles/dlib.dir/image_saver/save_webp.cpp.o
/home/aibrain/lib/dlib/dlib/image_saver/save_webp.cpp: In function ‘void dlib::impl::impl_save_webp(const string&, const uint8_t*, int, int, int, float, dlib::impl::webp_type)’:
/home/aibrain/lib/dlib/dlib/image_saver/save_webp.cpp:78:36: error: ‘WebPFree’ was not declared in this scope
                     WebPFree(output);
                                    ^
/home/aibrain/lib/dlib/dlib/image_saver/save_webp.cpp:86:28: error: ‘WebPFree’ was not declared in this scope
             WebPFree(output);
                            ^
dlib/CMakeFiles/dlib.dir/build.make:915: recipe for target 'dlib/CMakeFiles/dlib.dir/image_saver/save_webp.cpp.o' failed
make[2]: *** [dlib/CMakeFiles/dlib.dir/image_saver/save_webp.cpp.o] Error 1
CMakeFiles/Makefile2:97: recipe for target 'dlib/CMakeFiles/dlib.dir/all' failed
make[1]: *** [dlib/CMakeFiles/dlib.dir/all] Error 2
Makefile:155: recipe for target 'all' failed
make: *** [all] Error 2




## Steps to Reproduce
<!--- Provide an unambiguous set of steps to reproduce this problem.  What exactly do I have to type to reproduce your error?  If you don't post complete instructions you are much less likely to get help. -->
I installed by following the README documentation with the python 2.7.12 and cmake  3.24.1 on ubuntu 16.04

--- a/dlib/cmake_utils/find_libwebp.cmake
+++ b/dlib/cmake_utils/find_libwebp.cmake
@@ -47,6 +47,6 @@ if(WEBP_FOUND)
 
     if(NOT test_for_libwebp_worked)
         set(WEBP_FOUND 0)
-        message (STATUS "System copy of libwebp is broken.  Will build disable WebP support.")
+        message (STATUS "System copy of libwebp is either too old or broken.  Will disable WebP support.")
     endif()
 endif()

--- a/dlib/cmake_utils/find_libwebp.cmake
+++ b/dlib/cmake_utils/find_libwebp.cmake
@@ -13,22 +13,40 @@
 
 unset(WEBP_FOUND)
 
-FIND_PATH(WEBP_INCLUDE_DIR NAMES webp/decode.h)
+find_path(WEBP_INCLUDE_DIR NAMES webp/decode.h)
 
 if(NOT WEBP_INCLUDE_DIR)
     unset(WEBP_FOUND)
 else()
-    MARK_AS_ADVANCED(WEBP_INCLUDE_DIR)
+    mark_as_advanced(WEBP_INCLUDE_DIR)
 
     # Look for the library.
-    FIND_LIBRARY(WEBP_LIBRARY NAMES webp)
-    MARK_AS_ADVANCED(WEBP_LIBRARY)
+    find_library(WEBP_LIBRARY NAMES webp)
+    mark_as_advanced(WEBP_LIBRARY)
 
     # handle the QUIETLY and REQUIRED arguments and set WEBP_FOUND to TRUE if
     # all listed variables are TRUE
-    INCLUDE(${CMAKE_ROOT}/Modules/FindPackageHandleStandardArgs.cmake)
-    FIND_PACKAGE_HANDLE_STANDARD_ARGS(WebP DEFAULT_MSG WEBP_LIBRARY WEBP_INCLUDE_DIR)
+    include(${CMAKE_ROOT}/Modules/FindPackageHandleStandardArgs.cmake)
+    find_package_handle_standard_args(WebP DEFAULT_MSG WEBP_LIBRARY WEBP_INCLUDE_DIR)
 
-    SET(WEBP_LIBRARIES ${WEBP_LIBRARY})
-    SET(WEBP_INCLUDE_DIRS ${WEBP_INCLUDE_DIR})
+    set(WEBP_LIBRARIES ${WEBP_LIBRARY})
+    set(WEBP_INCLUDE_DIRS ${WEBP_INCLUDE_DIR})
+endif()
+
+if(WEBP_FOUND)
+    set(WEBP_TEST_CMAKE_FLAGS
+      "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}"
+      "-DCMAKE_INCLUDE_PATH=${CMAKE_INCLUDE_PATH}"
+      "-DCMAKE_LIBRARY_PATH=${CMAKE_LIBRARY_PATH}")
+
+    try_compile(test_for_libwebp_worked
+        ${PROJECT_BINARY_DIR}/test_for_libwebp_build
+        ${CMAKE_CURRENT_LIST_DIR}/test_for_libwebp
+        test_if_libwebp_is_broken
+        CMAKE_FLAGS "${WEBP_TEST_CMAKE_FLAGS}")
+
+    if(NOT test_for_libwebp_worked)
+        set(WEBP_FOUND 0)
+        message (STATUS "System copy of libwebp is broken.  Will build disable WebP support.")
+    endif()
 endif()

--- a/dlib/cmake_utils/test_for_libwebp/CMakeLists.txt
+++ b/dlib/cmake_utils/test_for_libwebp/CMakeLists.txt
@@ -1,0 +1,7 @@
+
+cmake_minimum_required(VERSION 3.8.0)
+project(test_if_libwebp_is_broken)
+
+include_directories(${WEBP_INCLUDE_DIR})
+add_executable(libwebp_test libwebp_test.cpp)
+target_link_libraries(libwebp_test ${WEBP_LIBRARY})

--- a/dlib/cmake_utils/test_for_libwebp/libwebp_test.cpp
+++ b/dlib/cmake_utils/test_for_libwebp/libwebp_test.cpp
@@ -1,0 +1,22 @@
+// Copyright (C) 2019  Davis E. King (davis@dlib.net), Nils Labugt
+// License: Boost Software License   See LICENSE.txt for the full license.
+
+#include <webp/encode.h>
+#include <webp/decode.h>
+#include <iostream>
+
+// This code doesn't really make a lot of sense.  It's just calling all the libjpeg functions to make
+// sure they can be compiled and linked.
+int main()
+{
+    std::cerr << "This program is just for build system testing.  Don't actually run it." << std::endl;
+    std::abort();
+
+    uint8_t* data;
+    size_t output_size = 0;
+    int width, height, stride;
+    float quality;
+    output_size = WebPEncodeRGB(data, width, height, stride, quality, &data);
+    WebPDecodeRGBInto(data, output_size, data, output_size, stride);
+    WebPFree(data);
+}

--- a/dlib/image_loader/webp_loader.cpp
+++ b/dlib/image_loader/webp_loader.cpp
@@ -6,9 +6,6 @@
 // only do anything with this file if DLIB_WEBP_SUPPORT is defined
 #ifdef DLIB_WEBP_SUPPORT
 
-#include "../array2d.h"
-#include "../pixel.h"
-#include "../dir_nav.h"
 #include "webp_loader.h"
 
 #include <webp/decode.h>

